docs/versioned_docs/version-0.15/infra/rabbitmq/index.md

```diff
@@ -4,4 +4,150 @@ description: "Producers and subscriptions for RabbitMQ"
+| `Priority`       | Message priority, from 0 to 9                                                    |
+| `RoutingKey`     | [Routing key][3] of the message, doesn't work with fan-out exchanges             |
+
+When the produced message has metadata, all metadata values will be converted to message headers. Subscriptions will restore headers back to metadata. If message metadata contains a correlation id (`eventuous.correlation-id` key), the value will be added to the RabbitMQ message correlation id property.
+
+## Subscriptions
+
+Eventuous supports consuming messages from RabbitMQ using [subscriptions](../../subscriptions).
+
+### Configuration
+
+As any other subscription, it can be added to the Di container using `AddSubscription` extension:
+
+```csharp
+builder.Services.AddSubscription<RabbitMqSubscription, RabbitMqSubscriptionOptions>(
+    "PaymentsIntegration",
+    builder => builder
+        .Configure(cfg => cfg.Exchange = "payments")
+        .AddEventHandler<PaymentsHandler>()
+);
+```
+
+The `Exchange` configuration property is mandatory as the subscription needs to know where it should consume messages from. Also, the subscription has a mandatory dependency on `ConnectionFactory`, so you'd need to register it in the container as described in the [producer configuration](#configuration) section above.
+
+For consuming messages, the subscription needs a queue bound to the specified exchange. By default, the subscription id is used as the queue name. You can override the queue name by specifying an alternative value using `QueueOptions.Queue` property of the subscription options.
+
+Besides the queue name, it's possible to configure the subscription with RabbitQ-specific parameters. Those include queue options, exchange options, and binding options. Eventuous provides default values for all those, so usually you would not need to change those. One option that you'd want to change si the concurrency limit value, which is set to `1` by default. As RabbitMQ doesn't guarantee message ordering anyway, you can speed up message processing by increasing the concurrency limit, so the subscription can consume messages in parallel. Eventuous will also adjust the prefetch count to accommodate for increased number of consumers, if necessary.
+
+As mention previously, RabbitMQ messages are published to an exchange, and consumed from a queue bound to that exchange. When the subscription starts, it makes sure that both the exchange and the queue exist, and the queue is bound to the exchange. If you start producing messages to an exchange created by the producer before starting the subscription at least once, and there's no queue and binding created upfront, those messages will be dropped. As long as the exchange has a binding to a subscription queue, the messages will be kept in the queue until consumed. Therefore, we recommend starting the subscription before producing messages.
```

Comment:
```suggestion
As mention edpreviously, RabbitMQ messages are published to an exchange, and consumed from a queue bound to that exchange. When the subscription starts, it makes sure that both the exchange and the queue exist, and the queue is bound to the exchange. If you start producing messages to an exchange created by the producer before starting the subscription at least once, and there's no queue and binding created upfront, those messages will be dropped. As long as the exchange has a binding to a subscription queue, the messages will be kept in the queue until consumed. Therefore, we recommend starting the subscriber before producing messages.
```
(qute possibly subscription is actually corrrect)

Reply:
Should be "as mentioned" :)

Reply:
fixed, lol!

Reply:
"Subscription" was correct, subscriber works too, it's just the terminology is "Subscription".